### ISSUE TITLE
Database is null or closed, not calling db.setLastSequence()

### ISSUE THREAD

**Issue by sapieneptus** [couchbase/couchbase-lite-java#64]
Running CBL java 1.1.0, we have an application which authenticates as a single user with all channels (`admin_channels : [ "*" ] ` ) and attempts to replicate essentially the entire database. About 2 seconds after our application starts, we get the error:

```
WARNING: Sync: com.couchbase.lite.replicator.ReplicationInternal$7@666c5994: Database is null or closed, not calling db.setLastSequence()  
```

after which, replication seems to stop. 

We have `Listener` and `ChangeTracker` logging enabled at `VERBOSE` level. Please advise what additional info we can provide. 

<hr>

Here is all of our CBL initialization code. There isn't much more to the app than this:

```
    private boolean setupManager(JSONObject config) {
        try {
            _manager = new Manager(new JavaContext("data"), Manager.DEFAULT_OPTIONS);
            Manager.enableLogging("ChangeTracker", Log.VERBOSE);
            Manager.enableLogging("Listener", Log.VERBOSE);
        } catch (IOException e) {
            e.printStackTrace();
            return false;
        }
        return true;
    }

    private Database.ChangeListener getChangeListener() {
        return new Database.ChangeListener() {
            public void changed(Database.ChangeEvent event) {
                Logger.log(new Date().toString() + ":: " + event.getChanges().size() + " changes");
                for (DocumentChange change : event.getChanges()) {
                    Logger.log("Doc " + change.getDocumentId() + " changed.");
                }
            }
        };
    }

    private boolean setupDatabase(JSONObject config) {
        try {
            _database = _manager.getDatabase((String)config.get(C.LOCAL_DATABASE_NAME_KEY));
        } catch (CouchbaseLiteException e) {
            e.printStackTrace();
            return false;
        }
        _database.addChangeListener(getChangeListener());
        return true;
    }

    private boolean setupPullReplication(JSONObject config) {
        String serverURL = config.get(C.SERVER_IP_KEY) + ":"
                            + config.get(C.SERVER_PORT_KEY) + "/"
                            + config.get(C.REMOTE_DATABASE_NAME_KEY) + "/" ;
        URL url = null;
        try {
            url = new URL("http://" +  serverURL); //TODO: handle https?
        } catch (MalformedURLException e) {
            e.printStackTrace();
            return false;
        }
        _pull = _database.createPullReplication(url);
        _pull.setContinuous(true);
        String user_id =  /* confidential */
        String password =  /* confidential */

        Logger.log("Authenticating with http://" + user_id + ":" + password + "@" + serverURL);
        Authenticator auth = new BasicAuthenticator(user_id, password);
        _pull.setAuthenticator(auth);
        _pull.start();
        return true;
    }
```

**Comment by hideki** [comment]
Hi @sapieneptus,

It seems database is not opened of some other code set null to db. Can you find any other error or warning message before it?
https://github.com/couchbase/couchbase-lite-java-core/blob/release/1.1.0/src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java#L736

maybe you can add following codes to enable more logs.
```java
            Manager.enableLogging(Log.TAG_SYNC_ASYNC_TASK, Log.VERBOSE);
            Manager.enableLogging(Log.TAG_SYNC, Log.VERBOSE);
            Manager.enableLogging(Log.TAG_DATABASE, Log.VERBOSE);
```

Thanks!

**Comment by sapieneptus** [comment]
Unfortunately, not  much has been revealed. Here's an example of a full log:

```
Aug 17, 2015 7:36:45 AM com.couchbase.lite.util.SystemLogger e
SEVERE: Sync: com.couchbase.lite.replicator.ReplicationInternal$4@2d671e3c checkSessionAtPath() response: {authentication_handlers=[default, cookie], ok=true, userCtx={channels={!=1, *=1345674}, name=user_55cfc5d89b81f04cf839608a}}
Mon Aug 17 07:36:47 UTC 2015:: 1 changes //this is our logging code
Doc team::team_Demo_Organization changed. //this is our logging code
Aug 17, 2015 7:36:48 AM com.couchbase.lite.util.SystemLogger w
WARNING: Sync: com.couchbase.lite.replicator.ReplicationInternal$7@ce6e2f2: Database is null or closed, not calling db.setLastSequence()
```

**Comment by hideki** [comment]
Hi @sapieneptus,

I can not find the log message from [the code](https://github.com/couchbase/couchbase-lite-java-core/blob/release/1.1.0/src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java#L704) before `WARNING: Sync: com.couchbase.lite.replicator.ReplicationInternal$7@ce6e2f2: Database is null or closed, not calling db.setLastSequence()` message .

Can you try to call ` Manager.enableLogging(...)` methods before creating Manager instance?

NOTE: I recommend to post this question on https://forums.couchbase.com/c/mobile too. Someone could answer for this.

Thanks,
Hideki

**Comment by sapieneptus** [comment]
That's odd, I enabled:

```
Manager.enableLogging(Log.TAG_CHANGE_TRACKER, Log.VERBOSE);
Manager.enableLogging(Log.TAG_LISTENER, Log.VERBOSE);
Manager.enableLogging(Log.TAG_SYNC_ASYNC_TASK, Log.VERBOSE);
Manager.enableLogging(Log.TAG_SYNC, Log.VERBOSE);
Manager.enableLogging(Log.TAG_DATABASE, Log.VERBOSE);
```

Maybe something's wrong with my gradle, let me try again.

**Comment by sapieneptus** [comment]
Oh *before* creating the manager - whoops, that's why. I'll report back

**Comment by farmazon3000** [comment]
Why this bug was closed? Is it fixed or what?

**Comment by hideki** [comment]
@farmazon3000,
We did not have enough information to investigate this issue. So I guess it was reason.
If you are facing this issue, please reopen the ticket.
Thanks!
